### PR TITLE
Refactor stage

### LINKED ISSUE #255: [refactoring] Reuse the logic for determining the scraper executable and yml files across cli.py and site.py
There are currently two code locations which do pretty much the same things:

* Location 1: [cli.py#_compute_has_fetch()](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/4458287104d03b71cc06dd375fe2b07b3d922f3c/vaccine_feed_ingest/cli.py#L99)
* Location 2: [ingest.py#run_fetch()](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/4458287104d03b71cc06dd375fe2b07b3d922f3c/vaccine_feed_ingest/stages/ingest.py#L19)

In both locations, the _things_ being done are the following:
* determine the executable (if there is one)
* determine the yml resource file to supply to the executable

**Proposal:**

Unify the associated logic so it can be reused across these two modules.

**Why?**

- DRY will prevent potential _drift_ in the future, leading to discrepancies.

**Context:**

I was responsible for introducing this code duplication while fixing #210 so I can also be responsible for fixing it :)

### PR DESCRIPTION
**Pull request by shashank025**
# 

| Key Details |
|-|
| Resolves #255 |
State:  |
Site:  |

## Notes

NA

## Data sample

Ran `all-stages` for `ak/arcgis`via:

```
 poetry run vaccine-feed-ingest all-stages ak/arcgis
```

and examined the first line of normalization output via:

```
head -1 out/ak/arcgis/normalized/2021-04-28T22\:44\:45/d92cbd6ff2524d7e92bef109f30cb366_0.parsed.normalized.ndjson
```

```json
{"id": "ak:arcgis:d92cbd6ff2524d7e92bef109f30cb366_0:85b07ac7-c9b6-4d09-83ec-a720998c2fbf", "name": "North Slope Borough Public Health Nursing-Wellness", "address": {"street1": "579 Kongosak St", "street2": null, "city": "Barrow", "state": "AK", "zip": "99723"}, "location": {"latitude": 71.29069998454801, "longitude": -156.7920099649541}, "contact": [{"contact_type": null, "phone": "(907) 852-0270", "website": null, "email": null, "other": null}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": {"drop_in": null, "appointments": true}, "inventory": [{"vaccine": "pfizer_biontech", "supply_level": null}], "access": null, "parent_organization": null, "links": null, "notes": ["NSB PHN Wellness clinic, please call for appointments 907-852-0270"], "active": null, "source": {"source": "ak:arcgis", "id": "85b07ac7-c9b6-4d09-83ec-a720998c2fbf", "fetched_from_uri": "https://services1.arcgis.com/WzFsmainVTuD5KML/ArcGIS/rest/services/COVID19_Vaccine_Site_Survey_API/FeatureServer/0", "fetched_at": "2021-04-29T05:44:55.676901", "published_at": "2020-12-25T13:00:00", "data": {"attributes": {"address": "579 Kongosak St", "city": "Barrow", "cost_other": null, "datesubmited": 1608930000000, "flu_acceptsInsurance": "yes_please_contact_your_insuran", "flu_cost": "no_cost", "flu_vaccinations": "pfizer", "flu_walkins": "no_please_make_an_appointment", "globalid": "85b07ac7-c9b6-4d09-83ec-a720998c2fbf", "needs_review": "approved", "objectid": 9, "phone": "907-852-0270", "publicEmail": null, "publicNotes": "NSB PHN Wellness clinic, please call for appointments 907-852-0270", "publicWebsite": null, "vaccinationSite": "North Slope Borough Public Health Nursing-Wellness", "zipcode": "99723"}, "geometry": {"spatialReference": {"latestWkid": 4326, "wkid": 4326}, "x": -156.7920099649541, "y": 71.29069998454801}}}}
```

## Before Opening a PR
- [x] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [x] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

File: vaccine_feed_ingest/stages/ingest.py [hunk 1]
```diff
@@ -283,6 +265,21 @@ def run_normalize(
     return True
 
 
+def resolve_executable(
```
Comment (bryanculbertson) [inline vaccine_feed_ingest/stages/ingest.py:268]: Can you move this helper into `site.py`?

**Review by bryanculbertson — APPROVED** [review]
Thank you for refactoring this!

**Review by bryanculbertson — CHANGES_REQUESTED** [review]
Can you move the helper into `site.py` and then I will merge this

**Comment by shashank025** [comment]
> Can you move the helper into `site.py` and then I will merge this

done!

**Comment by eliblock** [comment]
Hi! Sorry - it looks like this got swept up in the change for #278 . Could you pull in the latest `main` once more... then it seems like we'll be good to go?

**Comment by shashank025** [comment]
I've mostly worked with Mercurial at `$work`, so I'm not super familiar with PR merges, but it looks like this is now in a weird state; specifically, I intended only 3 files to be changed, but I see 22 files in the *Files Changed* section, and it seems like @bryanculbertson added some commits to this PR. Should I wait for one of you to merge my changes in, or would you like me to try and fix this branch?

**Comment by shashank025** [comment]
Ahh never mind, it looks like I need to fix some lint/test issues - let me do that first.

**Comment by eliblock** [comment]
@shashank025 hold off on fixing lint/test - since you are now modifying a bunch of files in this PR, some are in a weird state.

@bryanculbertson or I can help get this sorted out this evening. Sorry for the git confusion!

**Comment by shashank025** [comment]
> @shashank025 hold off on fixing lint/test - since you are now modifying a bunch of files in this PR, some are in a weird state.
> 
> @bryanculbertson or I can help get this sorted out this evening. Sorry for the git confusion!

Oops, I did another merge before  I read your reply. But I'll stop messing with this branch starting now!

**Comment by bryanculbertson** [comment]
@shashank025 I'll take a look at this in an hour and clean it up. Sorry for all of the confusiion

**Comment by eliblock** [comment]
This was tricky - I ended up moving the diff over to #299 rather than reducing this one.

Closing this PR in favor of that one.

**Pull request closed without merging**